price-feeder/oracle/client/client.go

```diff
@@ -105,33 +110,10 @@ func (r *passReader) Read(p []byte) (n int, err error) {
 	return n, err
 }
 
-// BroadcastPrevote attempts to broadcast a prevote transaction. Note, submiting
-// a prevote transaction does not require timeout functionality that vote
-// transactions require due to the fact that block timing validation exists on
-// the Umee node.
-//
+// BroadcastTx attempts to broadcast a transaction
+// a few times until it gets timed out
 // Ref: https://github.com/terra-money/oracle-feeder/blob/baef2a4a02f57a2ffeaa207932b2e03d7fb0fb25/feeder/src/vote.ts#L230
-func (oc OracleClient) BroadcastPrevote(msgs ...sdk.Msg) error {
-	ctx, err := oc.CreateContext()
-	if err != nil {
-		return err
-	}
-
-	factory, err := oc.CreateTxFactory()
-	if err != nil {
-		return err
-	}
-
-	return tx.BroadcastTx(ctx, factory, msgs...)
-}
-
-// BroadcastVote attempts to broadcast a vote transaction within the next voting
-// period.
-//
-// Ref: https://github.com/terra-money/oracle-feeder/blob/baef2a4a02f57a2ffeaa207932b2e03d7fb0fb25/feeder/src/vote.ts#L230
-func (oc OracleClient) BroadcastVote(nextBlockHeight int64, timeoutHeight int64, msgs ...sdk.Msg) error {
-	var height int64
-
+func (oc OracleClient) BroadcastTx(nextBlockHeight int64, timeoutHeight int64, msgs ...sdk.Msg) error {
```

Comment:
Needed to use this for both the voting and pre-voting because in every "voting period" there should be a vote and a pre-vote (in that order), and we can get errors here because of async transactions (e.g. getting back "account sequence mismatch, expected...")

Reply:
Shouldn't it be pre-vote and then vote, as voting reveals the salt and commitment to the exchange rates.

Reply:
Overall, yes, but not in each voting period. In the first voting period there will be only one transaction, a pre-vote, but in subsequent voting periods there will then be a vote, and then a pre-vote for the next voting period

| pre-vote | vote, pre-vote | vote, pre-vote | ... 

I'll clarify this in the comment